api/v1alpha1/node_order_types.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright (c) 2024 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+compliance with the License. You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is
+distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+implied. See the License for the specific language governing permissions and limitations under the
+License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NodeProvisionOrderSpec defines the desired state of an order to select and prepare a node that
+// will eventually be part of a deployment manager.
+type NodeProvisionOrderSpec struct {
+	// Location is the geographical location of the requested node.
+	//
+	// +kubebuilder:validation:Required
+	Location string `json:"location"`
+
+	// Extensions contains additional information that is associated to the order. It is up to
+	// the hardware manager to define and use these extensions.
+	//
+	// +kubebuilder:validation:Optional
+	Extensions map[string]string `json:"extensions,omitempty"`
+}
+
+// NodeProvisionOrderStatus describes the observed state of a node provisioning order.
+type NodeProvisionOrderStatus struct {
+	// Conditions represents the observations of the current state of the template. Possible
+	// values of the condition type are `Fulfilled` and `Failed`.
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	// BMC contains the details of the baseboard management controller of the node.
+	BMC BMC `json:"bmc"`
+}
+
+// BMC contains the details needed to connect to the baseboard management controller of a node.
+type BMC struct {
+	URL string `json:"url"`
```

Comment:
Just a reminder, we probably want a reference to the creds secret here.
E.g.
`
CredentialsSecretName string
`

Reply:
Yes, I added that in the latest version of the patch.